components/brave_wallet/browser/internal/orchard_shard_tree_manager_unittest.cc

```diff
@@ -0,0 +1,486 @@
+
+  for (int i = 0; i < 40; i++) {
+    std::optional<uint32_t> checkpoint;
+    if (i % 2 == 0) {
+      checkpoint = i * 2;
+    }
+
+    commitments.push_back(
+        CreateCommitment(CreateMockCommitmentValue(i, kDefaultCommitmentSeed),
+                         false, checkpoint));
+  }
+  OrchardTreeState orchard_tree_state;
+  auto result = CreateResultForTesting(orchard_tree_state, commitments);
+
+  tree_manager()->InsertCommitments(std::move(result));
+
+  EXPECT_EQ(10u, storage()->CheckpointCount(account_id()).value());
+  EXPECT_EQ(40u, storage()->MinCheckpointId(account_id()).value().value());
+  EXPECT_EQ(76u, storage()->MaxCheckpointId(account_id()).value().value());
+}
+
+TEST_F(OrchardShardTreeManagerTest, InsertWithFrontier) {
+  OrchardTreeState prior_tree_state;
+  prior_tree_state.block_height = 0;
+  prior_tree_state.tree_size = 48;
+  prior_tree_state.frontier = std::vector<uint8_t>(
+      {1,   72,  173, 200, 225, 47,  142, 44,  148, 137, 119, 18,  99,  211,
+       92,  65,  67,  173, 197, 93,  7,   85,  70,  105, 140, 223, 184, 193,
+       172, 9,   194, 88,  62,  1,   130, 31,  76,  59,  69,  55,  151, 124,
+       101, 120, 230, 247, 201, 82,  48,  160, 150, 48,  23,  84,  250, 117,
+       120, 175, 108, 220, 96,  214, 42,  255, 209, 44,  7,   1,   13,  59,
+       69,  136, 45,  180, 148, 18,  146, 125, 241, 196, 224, 205, 11,  196,
+       195, 90,  164, 186, 175, 22,  90,  105, 82,  149, 34,  131, 232, 132,
+       223, 15,  1,   211, 200, 193, 46,  24,  11,  42,  42,  182, 124, 29,
+       48,  234, 215, 28,  103, 218, 239, 234, 109, 10,  231, 74,  70,  197,
+       113, 131, 89,  199, 71,  102, 33,  1,   153, 86,  62,  213, 2,   98,
+       191, 65,  218, 123, 73,  155, 243, 225, 45,  10,  241, 132, 49,  33,
+       101, 183, 59,  35,  56,  78,  228, 47,  166, 10,  237, 50,  0,   1,
+       94,  228, 186, 123, 0,   136, 39,  192, 226, 129, 40,  253, 0,   83,
+       248, 138, 7,   26,  120, 212, 191, 135, 44,  0,   171, 42,  69,  6,
+       133, 205, 115, 4,   0,   0});
+
+  std::vector<OrchardCommitment> commitments;
+
+  commitments.push_back(
+      CreateCommitment(CreateMockCommitmentValue(48, kDefaultCommitmentSeed),
+                       false, std::nullopt));
+  commitments.push_back(
+      CreateCommitment(CreateMockCommitmentValue(49, kDefaultCommitmentSeed),
+                       false, std::nullopt));
+  commitments.push_back(
+      CreateCommitment(CreateMockCommitmentValue(50, kDefaultCommitmentSeed),
+                       true, std::nullopt));
+  commitments.push_back(CreateCommitment(
+      CreateMockCommitmentValue(51, kDefaultCommitmentSeed), false, 1));
+  commitments.push_back(
+      CreateCommitment(CreateMockCommitmentValue(52, kDefaultCommitmentSeed),
+                       false, std::nullopt));
+
+  auto result = CreateResultForTesting(prior_tree_state, commitments);
+  tree_manager()->InsertCommitments(std::move(result));
+
+  {
```

Comment:
is there a particular reason to have these blocks at the end of the test?

Reply:
Removed